System/Random.hs

```diff
@@ -430,22 +433,55 @@ mkStdGen32 sMaybeNegative = StdGen (s1+1) (s2+1)
 createStdGen :: Integer -> StdGen
 createStdGen s = mkStdGen32 $ fromIntegral s
 
-{- |
-With a source of random number supply in hand, the 'Random' class allows the
-programmer to extract random values of a variety of types.
 
-Minimal complete definition: 'randomR' and 'random'.
-
--}
+-- $uniform
+--
+-- Type classes 'Uniform' and 'UniformRange' are unrelated type
+-- classes. This is deliberate decision because there're types which
+-- could have instance for one type class but not the other.
+--
+-- For example: 'Integer', 'Float', 'Double' have instance for
+-- @UniformRange@ but there's no way to define @Uniform@.
+--
+-- Conversely there're types where @Uniform@ instance is possible
+-- while @UniformRange@ is not. One example is:
+--
+-- > instance (Uniform a, Uniform b) => Uniform (a,b)
+--
+-- This instance is very straightforward. @UniformRange@ isn't very
+-- sensible even if try to define one compatible with 'Ord' we'll run
+-- into problem that we don't know how many values inhabit @b@
+-- therefore we don't even know how many values we need to generate.
+--
+-- Another example is points on sphere cirle. Again @Uniform@ is
```

Comment:
I think adding `Complex` numbers here would be good for the example, since that's a concrete type in `base` for points on circle that doesn't have an `Ord` instance

Reply:
But `Complex` is case where you don't have neither Uniform nor UniformRange. There's no Uniform instance because domain is infinite and no UniformRange because there's no order. In order to gain Uniform one have to constrain it, say |z|=1

Reply:
That's right, I don't what I was thinking! (Guess what? Complex can have a `Random` instance ;P)

Reply:
I wonder what such instance could do? `random` generates |z|<1 & `randomR` interpolates between two points linearly?

Reply:
I was thinking something like this `random = mkPolar 1 uniformR (0, 2*pi)` could be pretty useful